src/components/material/digital-modal/DigitalModal.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useState } from "react";
+import { DigitalArticleService } from "../../../core/dbc-gateway/generated/graphql";
+import { useDplDasDigitalArticleOrderPOST } from "../../../core/dpl-cms/dpl-cms";
+import { useGetPatronInformationByPatronIdV2 } from "../../../core/fbs/fbs";
+import Modal from "../../../core/utils/modal";
+import { useText } from "../../../core/utils/text";
+import { Pid } from "../../../core/utils/types/ids";
+import DigitalModalBody from "./DigitalModalBody";
+import DigitalModalError from "./DigitalModalError";
+import DigitalModalSuccess from "./DigitalModalSuccess";
+import { createDigitalModalId } from "./helper";
+
+type DigitalModalProps = {
+  digitalArticleIssn: DigitalArticleService["issn"];
+  pid: Pid;
+};
+
+const DigitalModal: React.FunctionComponent<DigitalModalProps> = ({
+  digitalArticleIssn,
+  pid
+}) => {
+  const modalId = createDigitalModalId(digitalArticleIssn);
+  const t = useText();
+  const [email, setEmail] = useState<string>("");
+  const { mutate, isLoading, isSuccess, isError } =
+    useDplDasDigitalArticleOrderPOST();
+
+  const orderDigitalCopy = () => {
+    mutate({
+      data: {
+        pid,
+        email
+      }
+    });
+  };
+
+  // Pre fill the email field with the patron's email
+  const { data: patronData } = useGetPatronInformationByPatronIdV2();
+  useEffect(() => {
+    if (patronData?.patron?.emailAddress) {
```

Comment:
What happens if the service is slow and the user starts typing?

Reply:
That's a good question 😅 I would think that it will overwrite what the user is currently typing..

I think this will be a rare experience. I've tried throttling my network to slow 3g and it doesn't cause problems there

But let me know if you have a suggestion to improve it?

Reply:
That's a good question 😅 I would think that it will overwrite what the user is currently typing..

I think this will be a rare experience. I've tried throttling my network to slow 3g and it doesn't cause problems there

But let me know if you have a suggestion to improve it?